lib/postgrex/utils.ex

```diff
@@ -84,9 +84,18 @@ defmodule Postgrex.Utils do
     |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
     |> Keyword.update(:port, normalize_port(System.get_env("PGPORT")), &normalize_port/1)
     |> Keyword.put_new(:types, Postgrex.DefaultTypes)
+    |> default_app_name()
     |> Enum.reject(fn {_k, v} -> is_nil(v) end)
   end
 
+  @spec default_app_name(Keyword.t) :: Keyword.t
+  defp default_app_name(opts) do
+    app_name = System.get_env("PGAPPNAME") || "postgrex"
```

Comment:
We can get the OTP application name using `:application.get_application()` but it might be `:undefined` in some non-prod cases (i.e. spawning from shell). However I would prefer not to set a default application_name because it is not obvious how to disable it. Would an empty string be equivalent it to not being set? That would be an usual pattern.

Also if we want to be able to have default parameters I think we should have a single way to do it and should be be opt in. I think the place for that would be in application env for postgrex.